Vester/Tests/Host/Y-Add-Domain.Vester.ps1

```diff
@@ -0,0 +1,24 @@
+# Test file for the Vester module - https://github.com/WahlNetwork/Vester
+# Called via Invoke-Pester VesterTemplate.Tests.ps1
+
+# Test title, e.g. 'DNS Servers'
+$Title = 'Add Domain'
+
+# Test description: How New-VesterConfig explains this value to the user
+$Description = 'Add Domain'
+
+# The config entry stating the desired values
+$Desired = $cfg.host.domain
+
+# The test value's data type, to help with conversion: bool/string/int
+$Type = 'string'
+
+# The command(s) to pull the actual value for comparison
+
+[ScriptBlock]$Actual = {
+    (Get-VMHostAuthentication -VMHost $object).Domain
+}
+
+[ScriptBlock]$Fix = {
+   Get-VMHostAuthentication -VMHost $Object | Set-VMHostAuthentication -Domain $Desired -JoinDomain  -Confirm:$false -ErrorAction Stop
+}

```

Comment:
The name of this file starts with a "Y-" I think in an attempt to execute this portion last. You may want to review with Ashwin or Aditya why this is necessary and document that reasoning in the top of the file. Also, depending on that reasoning, we may want to name it something other than "Y-" but something that reflects the logic.